src/libs/Network.js

```diff
@@ -31,15 +35,21 @@ let didLoadPersistedRequests;
 Onyx.connect({
     key: ONYXKEYS.NETWORK_REQUEST_QUEUE,
     callback: (persistedRequests) => {
-        if (didLoadPersistedRequests || !persistedRequests) {
-            return;
-        }
+        ActiveClientManager.isReady.then(() => {
```

Comment:
Why are we using a promise to tell that the active client is ready? If there is a requirement for it to have such a state could we just set a boolean flag?

Reply:
I used to promise to overcome the race conditions. So when the user just landed on the page, this callback is fired before the client is set. And this won't repeat this to make sure that our queue is processed, I rely on promise.

Reply:
Ok thanks worries I have a different idea that I think will be more typical way to handle this like we do elsewhere.